### PR TITLE
fix: use container-aware thread count to avoid host op performance degradation in pods

### PR DESCRIPTION
**Pull request by rebel-jonghewk**
### 🚀 Summary of Changes

> *What does this PR do? What feature, fix, or improvement does it bring?*

---

### 📌 Related Issues / Tickets

* Resolves # `Jira Issue: SR-98`

---

## Problem

In container environments, `os.cpu_count()` reports all host CPUs (e.g., 96) while the pod is restricted to a cpuset subset (e.g., 32). Libraries like OpenMP, MKL, and NumPy use `os.cpu_count()` to spawn threads, causing massive thread contention on the limited cpuset — resulting in 30-60x slower host ops compared to bare metal on the same server.

Additionally, `set_cpu_affinity()` further narrows the K8s-assigned cpuset
(e.g., 32 → 16) by applying NUMA-aware physical-core-only selection on top
of an already-configured cpuset, which is unnecessary and harmful in
container environments.

## Solution

Detect container environments by comparing `os.sched_getaffinity(0)` with
`os.cpu_count()`. When a mismatch is detected:

- **Skip `set_cpu_affinity()`**: trust the K8s cpuset configuration
- **Set thread count to `allocated_cpus // 2`**: use physical cores only, excluding HT siblings
- **Set all thread pool env vars**: `OMP_NUM_THREADS`, `MKL_NUM_THREADS`,
  `OPENBLAS_NUM_THREADS`, `NUMEXPR_MAX_THREADS`, `RBLN_NUM_THREADS`

On bare metal, set NUMA-aware binding similar to rbln_worker.py (type2).

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

File: vllm_rbln/v1/worker/optimum_worker.py [hunk 1]
```diff
@@ -96,6 +99,41 @@ def __init__(
...
+            # Set all thread pool environment variables
+            os.environ["OMP_NUM_THREADS"] = str(num_threads)
+            os.environ["MKL_NUM_THREADS"] = str(num_threads)
+            os.environ["OPENBLAS_NUM_THREADS"] = str(num_threads)
+            os.environ["NUMEXPR_MAX_THREADS"] = str(num_threads)
+            os.environ["RBLN_NUM_THREADS"] = str(num_threads)
+
+            # Directly set PyTorch thread counts
+            torch.set_num_threads(num_threads)
+            torch.set_num_interop_threads(max(1, num_threads // 4))
```
Comment (rebel-eunji) [inline vllm_rbln/v1/worker/optimum_worker.py]: nit: Is it required?

**Comment by rebel-eunji** [comment]
@rebel-shshin I think this PR should be merged into dev, right?

File: vllm_rbln/v1/worker/optimum_worker.py (hunk 1 above)
Comment (rebel-jonghewk) [reply vllm_rbln/v1/worker/optimum_worker.py]: I'm not 100% sure. maybe we can delete it.

Comment (rebel-jonghewk) [inline vllm_rbln/v1/worker/optimum_worker.py]:
```suggestion
```

**Review by rebel-shshin — APPROVED** [review]
